test/Templates.Test/Helpers/AspNetProcess.cs

```diff
@@ -74,15 +74,26 @@ public AspNetProcess(ITestOutputHelper output, string workingDirectory, string p
             {
                 var dllPath = publish ? $"{projectName}.dll" : $"bin/Debug/{framework}/{projectName}.dll";
                 _process = ProcessEx.Run(output, workingDirectory, DotNetMuxer.MuxerPathOrDefault(), $"exec {dllPath}", envVars: envVars);
+                var listeningUrlString = GetListeningUrlString(output);
+                _listeningUri = new Uri(listeningUrlString, UriKind.Absolute);
             }
             else
             {
                 var exeFullPath = publish
                     ? Path.Combine(workingDirectory, $"{projectName}.exe")
                     : Path.Combine(workingDirectory, "bin", "Debug", framework, $"{projectName}.exe");
-                _process = ProcessEx.Run(output, workingDirectory, exeFullPath, envVars: envVars);
+                using (AddFirewallExclusion.Create(exeFullPath))
+                {
+                    _process = ProcessEx.Run(output, workingDirectory, exeFullPath, envVars: envVars);
```

Comment:
Assuming the firewall exclusion only needs to exist for the period where we're *starting* up the `.exe` and it *begins* the TCP listener, this looks great.